taskvine/src/manager/vine_file.c

```diff
@@ -143,6 +143,20 @@ struct vine_file *vine_file_clone(struct vine_file *f)
 	return f;
 }
 
+/* Make a URL reference to a file source*/
+
+const char *vine_file_make_file_url(const char *source)
+{
+
+	const char *abs_path = path_getcwd();
+
+	const char *result = string_format("file:%s/%s", abs_path, source);
+
+	free((void *)abs_path);
```

Comment:
Tricking the C to free a const is asking for trouble. I'm surprised the compiler let you. Anyway, abs_path shouldn't be const.